### PR TITLE
Socint 334

### PR DESCRIPTION
**Pull request by philwhiles**
EQ Launch refactored for reuse by RH, and updated to latest known EQ compliance.
CaseAPIClient has also been updated to reflect current RM API for telephone capture.

### CODE REVIEW

File: eq-launcher/src/main/java/uk/gov/ons/ctp/integration/eqlaunch/service/EqLaunchService.java [hunk 1]
```diff
@@ -1,9 +1,218 @@
...
+import uk.gov.ons.ctp.integration.eqlaunch.crypto.KeyStore;
+
+@Slf4j
+public class EqLaunchService {
+  private static final String ROLE_FLUSHER = "flusher";
+
+  // TODO : FLEXIBLE CASE
+  private Map<SurveyType, String[]> surveyTypeAddressAttribs = new HashMap<>();
+
+  private static final String[] socialAddressAttribs = {
```
Comment (rpcatling) [inline eq-launcher/src/main/java/uk/gov/ons/ctp/integration/eqlaunch/service/EqLaunchService.java:38]: conventional naming `SOCIAL_ADDRESS_ATTRIBS`

File: eq-launcher/src/test/resources/uk/gov/ons/ctp/integration/eqlaunch/service/TestEqLaunchService_payloadCreation.CaseUpdate.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+[
```
Comment (rpcatling) [inline eq-launcher/src/test/resources/uk/gov/ons/ctp/integration/eqlaunch/service/TestEqLaunchService_payloadCreation.CaseUpdate.json:1]: if you do the class rename suggested. these json files will need renaming

**Review by rpcatling — CHANGES_REQUESTED** [review]
Looks functionally OK I think.
A few minor comments.

File: eq-launcher/src/test/java/uk/gov/ons/ctp/integration/eqlaunch/service/TestEqLaunchService_payloadCreation.java [hunk 3]
```diff
@@ -202,6 +216,13 @@
    */
```
[the remark this answers is not in the document]
Comment (rpcatling) [reply eq-launcher/src/test/java/uk/gov/ons/ctp/integration/eqlaunch/service/TestEqLaunchService_payloadCreation.java:217]: You have fixed this one

File: case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientService.java [hunk 4]
```diff
@@ -66,61 +66,30 @@ public RmCaseDTO getCaseByCaseRef(Long caseReference, Boolean listCaseEvents) {
...
-  public SingleUseQuestionnaireIdDTO getSingleUseQuestionnaireId(
-      UUID caseId, boolean individual, UUID individualCaseId) {
+  public TelephoneCaptureDTO getSingleUseQuestionnaireId(
+      UUID caseId) {
     log.debug(
-        "getNewQuestionnaireIdForCase() calling Case Service to get new questionnaire ID",
-        kv("caseId", caseId),
-        kv("individual", individual),
-        kv("individualCaseId", individualCaseId));
+        "getSingleUseQuestionnaireId() calling Case Service to get new questionnaire ID",
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientService.java:73]: See above

File: case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientService.java [hunk 5]
```diff
@@ -66,61 +66,30 @@ public RmCaseDTO getCaseByCaseRef(Long caseReference, Boolean listCaseEvents) {
...
-            CASE_CREATE_SINGLE_USE_QUESTIONNAIRE_ID_PATH,
-            SingleUseQuestionnaireIdDTO.class,
+            CASE_GET_QUESTIONNAIRE_ID_PATH,
+            TelephoneCaptureDTO.class,
             null,
             queryParams,
             caseId.toString());
 
     log.debug(
         "getNewQuestionnaireIdForCase() generated new questionnaireId",
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientService.java:89]: fixed

File: case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/model/TelephoneCaptureDTO.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.ons.ctp.integration.caseapiclient.caseservice.model;
+
+import java.util.UUID;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import lombok.Data;
+
+/** This class holds data for a newly allocated questionnaire ID. i
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/model/TelephoneCaptureDTO.java:9]: fixed

File: case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientService.java [hunk 7]
```diff
@@ -66,61 +66,30 @@ public RmCaseDTO getCaseByCaseRef(Long caseReference, Boolean listCaseEvents) {
...
     // Ask Case Service to find case details
-    SingleUseQuestionnaireIdDTO newQuestionnaireId =
+    TelephoneCaptureDTO teleCaptureResponseDTO =
         caseServiceClient.getResource(
-            CASE_CREATE_SINGLE_USE_QUESTIONNAIRE_ID_PATH,
-            SingleUseQuestionnaireIdDTO.class,
+            CASE_GET_QUESTIONNAIRE_ID_PATH,
+            TelephoneCaptureDTO.class,
             null,
             queryParams,
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply case-api-client/src/main/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientService.java:85]: null used instead now

File: case-api-client/src/test/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientServiceTest.java [hunk 8]
```diff
@@ -54,66 +53,36 @@ public void testGetCaseById_withNoCaseEvents() {
...
-    // Run the request
-    QuestionnaireIdDTO results = caseServiceClientService.getReusableQuestionnaireId(testUuid);
-
-    assertEquals(resultsFromCaseService.getQuestionnaireId(), results.getQuestionnaireId());
-    assertEquals(resultsFromCaseService.getFormType(), results.getFormType());
-    assertEquals(resultsFromCaseService.isActive(), results.isActive());
-  }
-
   @Test
   public void doTestGetSingleUseQid() throws Exception {
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply case-api-client/src/test/java/uk/gov/ons/ctp/integration/caseapiclient/caseservice/CaseServiceClientServiceTest.java:57]: fair enough

File: eq-launcher/src/test/java/uk/gov/ons/ctp/integration/eqlaunch/service/TestEqLaunchService_payloadCreation.java [hunk 9]
```diff
@@ -1,11 +1,25 @@
...
+import uk.gov.ons.ctp.common.domain.Language;
+import uk.gov.ons.ctp.common.domain.Source;
+import uk.gov.ons.ctp.common.domain.SurveyType;
+import uk.gov.ons.ctp.common.event.model.CaseUpdate;
+import uk.gov.ons.ctp.common.event.model.CollectionExerciseUpdate;
+import uk.gov.ons.ctp.common.event.model.UacUpdate;
+import uk.gov.ons.ctp.integration.eqlaunch.crypto.JweDecryptor;
+import uk.gov.ons.ctp.integration.eqlaunch.crypto.KeyStore;
 
 public class TestEqLaunchService_payloadCreation {
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply eq-launcher/src/test/java/uk/gov/ons/ctp/integration/eqlaunch/service/TestEqLaunchService_payloadCreation.java:24]: renamed

File: eq-launcher/src/main/java/uk/gov/ons/ctp/integration/eqlaunch/service/EqLaunchService.java (hunk 1 above)
Comment (philwhiles) [reply eq-launcher/src/main/java/uk/gov/ons/ctp/integration/eqlaunch/service/EqLaunchService.java:38]: renamed

**Review by rpcatling — APPROVED** [review]

**Pull request merged**